### PR TITLE
Mech Blue R1 dual channel solenoid

### PR DESCRIPTION
**Pull request by SiddharthKadari**
Fixed R1 to successfully retract the Dual Channel Solenoids. Previous R1 could only extend a Dual Channel Solenoid.

### CODE REVIEW

File: src/main/java/frc/robot/systems/BallHandlingFSM.java [hunk 1]
```diff
@@ -62,6 +66,7 @@ public FSMState getCurrentState() {
 	public void reset() {
 		currentState = FSMState.IDLE;
 
+		pushCommandTime = -1;
```
Comment (tsoim) [inline src/main/java/frc/robot/systems/BallHandlingFSM.java:69]: magic num? why `-1`?

File: src/main/java/frc/robot/systems/BallHandlingFSM.java [hunk 2]
```diff
@@ -100,20 +108,24 @@ public void update(TeleopInput input) {
...
 					return FSMState.IDLE;
 				}
 
 			case FIRING:
-				if (Timer.getFPGATimestamp() - pushCommandTime > PUSH_TIME_SECONDS) {
-					return FSMState.IDLE;
-				} else {
-					return FSMState.FIRING;
-				}
+				return FSMState.IDLE;
```
Comment (tsoim) [inline src/main/java/frc/robot/systems/BallHandlingFSM.java:125]: doesn't this mean you'll always just return to IDLE immediately from FIRING/RETRACTING and set both solenoids to false? or am I understanding this wrong. would love to see an updated FSM

File: src/main/java/frc/robot/systems/BallHandlingFSM.java (hunk 1 above)
Comment (SiddharthKadari) [reply src/main/java/frc/robot/systems/BallHandlingFSM.java:69]: We use pushCommandTime == -1 as an indicator that the solenoid is currently retracted.

File: src/main/java/frc/robot/systems/BallHandlingFSM.java (hunk 2 above)
Comment (SiddharthKadari) [reply src/main/java/frc/robot/systems/BallHandlingFSM.java:125]: Yes this is correct. The reason we do this, is that in order to set the direction of the solenoid, we must pulse the solenoid in that direction for only a few milliseconds, then power should be turned off, as it is undesired for the solenoid to constantly be receiving unnecessary power.

**Comment by AG6GR** [comment]
Is this pull request still required? Given that you have #9 also open? In the future you can push new commits to the same branch to update an existing PR.

**Comment by SiddharthKadari** [comment]
> Is this pull request still required? Given that you have #9 also open? In the future you can push new commits to the same branch to update an existing PR.

You are correct this PR is no longer required, sorry about that redundancy.

**Pull request closed without merging**